sdks/java/io/hadoop/input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -845,7 +743,8 @@ public Double getFractionConsumed() {
        */
       private Double getProgress() throws IOException, InterruptedException {
         try {
-          return (double) recordReader.getProgress();
+          float progress = recordReader.getProgress();
+          return (double) progress < 0 || progress > 1 ? 0.0 : progress;
```

Comment:
This means that when we can't determine progress we're returning 0 in getFractionConsumed

However, Bounded source states "or {@code null} if such an estimate is not available."

perhaps we can change getFractionConsumed to return null in cases where the value is 0 since we can't store a null in progressValue? Or we could use a signal value like -1.0 to indicate "not known" and then return null from getFractionConsumed (that might result in errors down the line)

We should add a test to check that this is working correctly (returning null when recordReader.getProgress() returns nonsense)